src/ray/core_worker/test/direct_task_transport_test.cc

```diff
@@ -0,0 +1,266 @@
+#include "gtest/gtest.h"
+
+#include "ray/common/task/task_spec.h"
+#include "ray/core_worker/store_provider/memory_store/memory_store.h"
+#include "ray/core_worker/store_provider/memory_store_provider.h"
+#include "ray/core_worker/transport/direct_task_transport.h"
+#include "ray/raylet/raylet_client.h"
+#include "ray/rpc/worker/core_worker_client.h"
+#include "src/ray/util/test_util.h"
+
+namespace ray {
+
+class MockWorkerClient : public rpc::CoreWorkerClientInterface {
+ public:
+  ray::Status PushTaskImmediate(
+      std::unique_ptr<rpc::PushTaskRequest> request,
+      const rpc::ClientCallback<rpc::PushTaskReply> &callback) override {
+    callbacks.push_back(callback);
+    return Status::OK();
+  }
+
+  std::vector<rpc::ClientCallback<rpc::PushTaskReply>> callbacks;
+};
+
+class MockRayletClient : public WorkerLeaseInterface {
+ public:
+  ray::Status ReturnWorker(int worker_port) {
+    num_workers_returned += 1;
+    return Status::OK();
+  }
+
+  ray::Status RequestWorkerLease(const ray::TaskSpecification &resource_spec) {
+    num_workers_requested += 1;
+    return Status::OK();
+  }
+
+  int num_workers_requested = 0;
+  int num_workers_returned = 0;
+};
+
+TEST(LocalDependencyResolverTest, TestNoDependencies) {
```

Comment:
Can you add a line or two to each test to describe it?

Reply:
Done